docs(onmoduleinit): add info about onmoduleinit execution order (#2447)
nestjs/docs.nestjs.com: content/fundamentals/lifecycle-events.md

> warning **Warning** The lifecycle hooks listed above are not triggered for **request-scoped** classes. Request-scoped classes are not tied to the application lifecycle and their lifespan is unpredictable. They are exclusively created for each request and automatically garbage-collected after the response is sent.

> > info **Info** Execution order of `onModuleInit()` and `onApplicationBootstrap()` directly depends on the order of module imports, awaiting previous hook. If services are imported via providers option then `onModuleInit()` and `onApplicationBootstrap()` will be called simultaneously.

Reviewer: ```suggestion
> info **Hint** Execution order of `onModuleInit()` and `onApplicationBootstrap()` directly depends on the order of module imports, awaiting the previous hook.
```
